src/java_bytecode/java_object_factory.cpp

```diff
@@ -181,10 +512,9 @@ Function: new_tmp_symbol
 
 \*******************************************************************/
 
-namespace {
-symbolt &new_tmp_symbol(symbol_tablet &symbol_table)
+symbolt &new_tmp_symbol(symbol_tablet &symbol_table, const std::string& prefix)
 {
-  static int temporary_counter=0;
+  static size_t temporary_counter=0; // TODO change this
```

Comment:
I changed to `// TODO encapsulate as class variable` I think the problem is that it's used outside of `java_object_factoryt`, which is the place it belongs most likely.